stlab/concurrency/future.hpp

```diff
@@ -296,20 +297,20 @@ struct shared_base<T, enable_if_copyable<T>> : std::enable_shared_from_this<shar
     template <typename F, typename... Args>
     void set_value(F& f, Args&&... args);
 
-    auto get_try() -> boost::optional<T> {
+    auto get_try() -> boost::optional<const T&> {
```

Comment:
std::optional<> doesn't allow references because an optional reference to a temporary wouldn't do lifetime extension.  Boost worked around this issue in later releases (not sure which version) but disallowing assigning an rvalue to an optional reference. But I'm not sure there are any papers for std::optional to pick-up this change. We could add a private call to get the value when we know it is ready (for use in then() ) that just return a const T& (not an optional). That would be a bit more efficient. Not sure how we want to deal with get_try() though - leave it as a boost::optional reference or ???.